### ISSUE TITLE
Option to trigger a11y checks automatically based on selected page events

### ISSUE THREAD

**Issue by mohanraj-r** [microsoft/accessibility-insights-web#2209]
## Is your feature request related to a problem? Please describe.
Right now the accessibility validation through the browser extension takes multiple clicks to trigger. 

## Describe the desired outcome
An automatic mode in which the chosen check (e.g. Fastpass) is run automatically on selected page events (e.g. page load) and to optionally highlight issues (e.g. based on a keyboard shortcut) and optional whitelist/blacklist of URLs/domains. The number of issues found could be displayed as a badge on the extension. This would enable a relatively hands-free way of triggering accessibility checks for pages under development and viewing their results.

## Describe alternatives you've considered
NA

## Additional context
This option could be useful in automated testing environments - where screenshots / recordings of the tests are done esp on failure. This might require export/import settings - or ability to configure extension in an automated way. This could take it a step closer to https://github.com/microsoft/accessibility-insights-web/issues/787

**Comment by ferBonnin** [comment]
Hi @mohanraj-r thanks for the feature request!

One question, have you take a look at the [Keyboard Shortcuts for Accessibility Insights for Web](https://accessibilityinsights.io/docs/en/web/reference/keyboard)?

**Comment by mohanraj-r** [comment]
I guess using keyboard shortcuts could be useful (that is a cool feature, thanks for the pointer). But the way I see it, this option could also be useful in automated testing environments - where screenshots / recordings of the tests are done esp on failure or even when just manually running a UI test with the extension installed and watching it.  

For automated environments though this might require 
* export/import settings - or ability to configure extension in an automated way
* and the ability to also add a brief visible summary to the highlighted issues rather than just exposing them on clicking the highlighted issue
This could take it a step closer to https://github.com/microsoft/accessibility-insights-web/issues/787

But even otherwise, think having the option to automatically highlight might make it easier than pressing the keyboard shortcuts every time in a local dev workflow.

**Comment by ferBonnin** [comment]
hi @mohanraj-r got it! for the scenario you are talking about in automated check environments, please take a look at the [axe-pipelines-samples ](https://github.com/microsoft/axe-pipelines-samples) repo.

**Comment by mohanraj-r** [comment]
@ferBonnin looking at [axe-pipelines-samples/index.test.ts](https://github.com/microsoft/axe-pipelines-samples/blob/master/typescript-selenium-webdriver-sample/tests/index.test.ts) it looks like they are not using the accessibility insights project? Looks like they are distinct efforts with no reuse? I am looking/hoping for reusing the same workflow based on the extension in an automated testing workflow.

**Comment by ferBonnin** [comment]
> @ferBonnin looking at [axe-pipelines-samples/index.test.ts](https://github.com/microsoft/axe-pipelines-samples/blob/master/typescript-selenium-webdriver-sample/tests/index.test.ts) it looks like it is not using the accessibility insights project? Looks like it is a distinct effort with no reuse? I am looking/hoping for reusing the same workflow based on the extension in an automated testing workflow.
This project uses the same automated checks (axe-core) we use in Accessibility Insights for Web; since its for an automated check environment, the workflow and what you can test (in this case just automated checks) is different.

**Comment by mohanraj-r** [comment]
I understand that the vision I have is probably not aligned with the goals/plans of this project. Hence closing the issue.

**Comment by ferBonnin** [comment]
@mohanraj-r from what we understand, this feature request is about running the checks automatically on a page event instead of waiting on a user action. This would be helpful for automated environments. is our understanding of the request correct?

if yes, there are two options we currently have:

- using shortcuts in Accessibility Insights for Web
- using the axe-pipelines-samples, to run the automated checks in an automated environment. This leverages the same automated checks we use in the extension.

Having automatically run a scan when a page event happens is currently not possible: for additional security and privacy, our extension requires the user to initiate the action and will not start running automatically (more information in [release V2.13.1](https://github.com/microsoft/accessibility-insights-web/releases/tag/v2.13.1)). Changing that behaviour to provide a fully automatic experience on certain page events would require more aggressive extension permissions and that would impact the user experience and the velocity in which we release.

Please let us know if this makes sense and if the two options above help with this request. We are also always happy to talk more about this.
